engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/CatchEventTransformer.java

```diff
@@ -133,4 +138,16 @@ private void transformErrorEventDefinition(
     final var executableError = context.getError(error.getId());
     executableElement.setError(executableError);
   }
+
+  public void transformLinkEventDefinition(
+      final TransformContext context,
+      final ExecutableCatchEventElement executableElement,
+      final LinkEventDefinition linkEventDefinition) {
+
+    final var process = context.getCurrentProcess();
+    process.attach(executableElement);
```

Comment:
:x: We don't need to attach the catch event anymore. Instead, we bind the link events directly.

We should remove this code in the transformer, including the `attach()` method in the `ExecutableActivity`.